public/app/features/integrated-alerting/components/Table/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { FC, useState, useMemo, useEffect } from 'react';
+import { useStyles, IconName, Button, Select } from '@grafana/ui';
+import { PaginationProps } from './Pagination.types';
+import { getStyles } from './Pagination.styles';
+import { Messages } from './Pagination.messages';
+
+export const Pagination: FC<PaginationProps> = ({
+  pageCount: propPageCount,
+  initialPageIndex,
+  pageSize,
+  pagesPerView,
+  nrRowsOnCurrentPage,
+  totalItems,
+  pageSizeOptions,
+  onPageChange,
+  onPageSizeChange,
+}) => {
+  // Zero pages probably won't make the pagination show up, but here we should be agnostic to that
+  const [pageCount, setPageCount] = useState(Math.max(propPageCount, 1));
```

Comment:
Can you explain this? I don't really understand why this is needed

Reply:
Just to prevent having no page buttons at all, in case `pageCount` from props comes as 0.

Reply:
The way we have it now, you won't even see the pagination in case there's no results (only that warning message). But I believe the component here shouldn't be aware of that

Reply:
Well, are we supposed to see the pagination if there are no pages to browse? :) If this wasn't a requirement, I would confirm on this. We should stick to YAGNI :)

Reply:
The requirement for this particular page is that there's no pagination without data. However, the pagination component itself can live in any other table, including one that might show the pagination even without data.
Even with zero pages, it should always show `| << | < | 1 | > | >> |`.  `| << | < | > | >> |` is something you don't see out there I guess.

Reply:
Not sure I'm following you. So the requirement says that there shouldn't be any pagination, right? The fact that the component should be generic and can be used in pages where we might want to show the paginator even with no data, is not compliant with YAGNI.

The components we build can be made generic if need be, but they shouldn't be created always as generic components.

Reply:
I got it. Removed the safety check.